Fix compatibility issues with upgraded rb_libtorrent

Patch by Gordon Bos

--- src/Downloader.h
+++ src/Downloader.h
@@ -30,6 +30,7 @@
 #include <config.h>
 #endif
 
+#include <unistd.h>
 #include <string>
 #include <map>
 #include <list>
--- src/TorrentDownloader.cpp
+++ src/TorrentDownloader.cpp
@@ -446,7 +446,7 @@
 
 		torrent_info ti=this->handle.get_torrent_info();
 		for(torrent_info::file_iterator fIt=ti.begin_files();fIt!=ti.end_files();fIt++){
-			string filename=(*fIt).path.native_file_string();
+			string filename=ti.files().file_path(*fIt);
 
             boost::filesystem::path p(filename);
             if(boost::filesystem::exists(p)) {
@@ -1026,7 +1026,8 @@
 
 	this->s.set_settings(settings);
 
-	if(!this->s.listen_on(make_pair(startport,endport))){
+	this->s.listen_on(make_pair(startport,endport));
+	if(!this->s.is_listening()){
 		syslog(LOG_CRIT,"Could not open a listening port.");
 		throw std::runtime_error("Could not open a listening port.");
 	}
--- src/filetransferdaemon.cpp
+++ src/filetransferdaemon.cpp
@@ -134,7 +134,7 @@
 int main(int argc,char** argv){
 	int daemonize=0;
 	int version=0;
-	const char* group="www-data";
+	const char* group="apache";
 
 	try {
 		openlog("ftd",LOG_PERROR,LOG_DAEMON);
